k8s/k8s.go

```diff
@@ -180,11 +182,22 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, nodename string, calicoCl
 		case ipAddrs == "" && ipAddrsNoIpam == "":
 			// Call IPAM plugin if ipAddrsNoIpam or ipAddrs annotation is not present.
 			logger.Debugf("Calling IPAM plugin %s", conf.IPAM.Type)
-			result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
+			res, err := ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
 			if err != nil {
 				return nil, err
 			}
-			logger.Debugf("IPAM plugin returned: %+v", result)
+			logger.Debugf("IPAM plugin returned: %+v", res)
+
+			// Convert IPAM result into current Result
+			result, err = current.NewResultFromResult(res)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(result.IPs) == 0 {
+				return nil, errors.New("IPAM plugin returned missing IP config")
```

Comment:
as discussed f2f, it is the same (just different import name because of conflict in calico.go)